src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/ReadThreadingAssembler.java

```diff
@@ -454,25 +484,48 @@ private AssemblyResult getAssemblyResult(final Haplotype refHaplotype, final int
 
         printDebugGraphTransform(rtgraph, refHaplotype.getLocation() + "-sequenceGraph." + kmerSize + ".0.1.cleaned_readthreading_graph.dot");
 
-        final SeqGraph initialSeqGraph = rtgraph.toSequenceGraph();
-        if (debugGraphTransformations) {
-            initialSeqGraph.printGraph(new File(debugGraphOutputPath, refHaplotype.getLocation() + "-sequenceGraph." + kmerSize + ".0.1.initial_seqgraph.dot"), 10000);
-        }
+        // Either return an assembly result with a sequence graph or with an unchanged sequence graph deptending on the kmer duplication behavior
+        if (generateSeqGraph) {
+            final SeqGraph initialSeqGraph = rtgraph.toSequenceGraph();
+            if (debugGraphTransformations) {
+                rtgraph.printGraph(new File(debugGraphOutputPath, refHaplotype.getLocation() + "-sequenceGraph." + kmerSize + ".0.1.initial_seqgraph.dot"), 10000);
+            }
 
-        // if the unit tests don't want us to cleanup the graph, just return the raw sequence graph
-        if ( justReturnRawGraph ) {
-            return new AssemblyResult(AssemblyResult.Status.ASSEMBLED_SOME_VARIATION, initialSeqGraph, null);
-        }
+            // if the unit tests don't want us to cleanup the graph, just return the raw sequence graph
+            if (justReturnRawGraph) {
+                return new AssemblyResult(AssemblyResult.Status.ASSEMBLED_SOME_VARIATION, initialSeqGraph, null);
+            }
 
-        if ( debug ) {
-            logger.info("Using kmer size of " + rtgraph.getKmerSize() + " in read threading assembler");
-        }
-        printDebugGraphTransform(initialSeqGraph, refHaplotype.getLocation() + "-sequenceGraph." + kmerSize + ".0.2.initial_seqgraph.dot");
-        initialSeqGraph.cleanNonRefPaths(); // TODO -- I don't this is possible by construction
+            if (debug) {
+                logger.info("Using kmer size of " + rtgraph.getKmerSize() + " in read threading assembler");
+            }
+            printDebugGraphTransform(initialSeqGraph, refHaplotype.getLocation() + "-sequenceGraph." + kmerSize + ".0.2.initial_seqgraph.dot");
+            initialSeqGraph.cleanNonRefPaths(); // TODO -- I don't this is possible by construction
```

Comment:
Specify why you don't think it's possible.

Reply:
That code was copied from above. The reason I suspect when that was added is because dangling paths have already been pruned. I believe at least some variation of this method is still necessary here however because there are destructive pruning steps taking before this (also edge multiplicity might be changed at places in the seq graph transformations).

Reply:
Put this in the code then.  Even if you're not 100% certain, it's much better than nothing.